assets/src/application/ui/layout/modal/ModalWithAlert.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect } from 'react';
+import { useDisclosure } from '@chakra-ui/core';
+import { __ } from '@wordpress/i18n';
+import { canUseDOM } from '@appServices/utilities/dom';
+import { ESCAPE } from '@wordpress/keycodes';
+
+import { AlertDialog } from '@infraUI/display';
+import { Button as ButtonAdapter } from '@infraUI/inputs';
+import { Modal, ModalProps } from '@infraUI/layout/modal';
+
+interface Props extends ModalProps {
+	cancelBtnText?: string;
+	header?: string;
+	okBtnText?: string;
+	showAlertOnEscape: boolean;
+}
+
+const ModalWithAlert: React.FC<Props> = ({ children, showAlertOnEscape, ...props }) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+	const cancelRef = React.useRef();
+	const cancelBtnText = props.cancelBtnText || __('No');
+	const header = props.header || __('Are you sure you want to close this?');
+	const okBtnText = props.okBtnText || __('Yes');
+	const onEscape = ({ keyCode }) => keyCode === ESCAPE && onOpen();
```

Comment:
Not good for TS, try to set the return type of the function to see the problem with that magical shortcut 😄

Reply:
agree, made it TS-friendly:
```
	const onEscape = ({ keyCode }): void => {
		if (keyCode === ESCAPE) {
			onOpen();
		}
	};
```
***
was in one-liner mood before :smiley: